service-base/src/test/java/org/eclipse/hono/service/tenant/TenantMessageFilterTest.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.service.tenant;
+
+import static org.eclipse.hono.util.MessageHelper.APP_PROPERTY_RESOURCE;
+import static org.eclipse.hono.util.MessageHelper.APP_PROPERTY_TENANT_ID;
+import static org.eclipse.hono.util.TenantConstants.Action;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.util.*;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonHelper;
+
+/**
+ * Test verifying that the filter complies with the tenant message format specification.
+ */
+public class TenantMessageFilterTest {
+
+    private static final String DEFAULT_TENANT = "DEFAULT_TENANT";
+
+    @Test
+    public void testVerifyDetectsMissingTenantProperty() {
+        // GIVEN a valid tenant GET message without an AMQP value
+        final Message msg = givenAMessageHavingProperties(Action.ACTION_GET);
+        // WHEN receiving the message via a link with any tenant
+        final ResourceIdentifier linkTarget = getResourceIdentifier(DEFAULT_TENANT);
+
+        // THEN message validation fails
+        assertFalse(TenantMessageFilter.verify(linkTarget, msg));
+    }
+
+    @Test
+    public void testVerifySucceedsForValidGetAction() {
+        // GIVEN a tenant GET message for tenant DEFAULT_TENANT
+        final Message msg = givenAMessageHavingProperties(Action.ACTION_GET);
+        MessageHelper.addProperty(msg, APP_PROPERTY_TENANT_ID, DEFAULT_TENANT);
+        // WHEN receiving the message via a link with matching target address
+        final ResourceIdentifier linkTarget = getResourceIdentifier(DEFAULT_TENANT);
+
+        // THEN message validation succeeds
+        assertTrue(TenantMessageFilter.verify(linkTarget, msg));
+        assertMessageAnnotationsContainProperties(msg, DEFAULT_TENANT);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testTenantMessageForEventBus() {
+        Message msg = ProtonHelper.message();
+        msg.setSubject(Action.ACTION_GET.toString());
+        MessageHelper.addTenantId(msg, Constants.DEFAULT_TENANT);
+
+        ResourceIdentifier resource = ResourceIdentifier.from(TenantConstants.TENANT_ENDPOINT, Constants.DEFAULT_TENANT, null);
+        MessageHelper.annotate(msg, resource);
+
+        final JsonObject tenantMsg = TenantConstants.getTenantMsg(msg);
```

Comment:
this seems to be a unit test for class `TenantConstants`

Reply:
Correct - and again `CredentialsMessageFilterTest` affected as well. Will fix both.